odoo/models.py

```diff
@@ -5827,14 +5827,16 @@ def __contains__(self, item):
 
                 any(item == record for record in self)
         """
-        if isinstance(item, BaseModel) and self._name == item._name:
-            return len(item) == 1 and item.id in self._ids
-        elif isinstance(item, str):
-            return item in self._fields
-        elif isinstance(item, BaseModel):
-            raise TypeError(f"cannot compare different models: '{self._name}()' and '{item._name}()'")
-        else:
-            raise TypeError(f"unsupported operand type(s) for \"in\": '{self._name}()' and '{type(item)}'")
+        try:
+            if self._name == item._name:
+                return len(item) == 1 and item.id in self._ids
+            else:
+                raise TypeError(f"cannot compare different models: '{self._name}()' and '{item._name}()'")
```

Comment:
Because of the `return`, one can simplify the code and avoid the `else` line:
```suggestion
            if self._name == item._name:
                return len(item) == 1 and item.id in self._ids
            raise TypeError(f"cannot compare different models: '{self._name}()' and '{item._name}()'")
```
The same can be done in all changes.